packages/web-components/src/components/date-picker/__tests__/date-picker-test.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright IBM Corp. 2019, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, render } from 'lit';
+
+describe('cds-date-picker', () => {
+  describe('Invalid and Warning States with Disabled/ReadOnly', () => {
+    it('should not show invalid state when disabled', async () => {
+      render(
+        html`
+          <cds-date-picker-input
+            invalid
+            invalid-text="Invalid date"
+            disabled
+            label-text="Date Input"></cds-date-picker-input>
+        `,
+        document.body
+      );
+      await Promise.resolve();
```

Comment:
```suggestion
      const el = await fixture(
        html`
          <cds-date-picker-input
            invalid
            invalid-text="Invalid date"
            disabled
            label-text="Date Input"></cds-date-picker-input>`
            );
    await el.updateComplete;
```

`await fixture` should be used instead, and `document.body` may not be necessary.

See other test files for coding conventions/practices:
- [One](https://github.com/carbon-design-system/carbon/blob/main/packages/web-components/src/components/slider/__tests__/slider-test.js)
- [Two](https://github.com/carbon-design-system/carbon/blob/main/packages/web-components/src/components/time-picker/__tests__/time-picker-test.js)
- [Three](https://github.com/carbon-design-system/carbon/blob/main/packages/web-components/src/components/password-input/__tests__/password-input-test.js)